51_fl_diff_shipment.py

```diff
@@ -0,0 +1,258 @@
+)
+other_states_shipment_copy
+# calculating and displaying results for drug related deaths - year, state and county wise
+
+other_states_result = (
+    other_states_shipment_copy.groupby(["Year", "StateName", "county_test"])[
+        "shipment_per_100k"
+    ]
+    .sum()
+    .reset_index()
+)
+other_states_result.head()
+# calculating yearwise drug related deaths results for Alabama
+alabama_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Alabama"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+alabama_results
+# calculating yearwise drug related deaths results for Georgia
+georgia_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Georgia"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+georgia_results
+# calculating yearwise drug related deaths results for Mississippi
+mississippi_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Mississippi"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+mississippi_results
+# calculating yearwise drug related deaths results for South Carolina
+south_carolina_results = (
+    other_states_shipment_copy[
+        other_states_shipment_copy["StateName"] == "South Carolina"
+    ]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+south_carolina_results
+# calculating yearwise drug related deaths results for Tennessee
+tennessee_results = (
+    other_states_shipment_copy[other_states_shipment_copy["StateName"] == "Tennessee"]
+    .groupby("Year")["shipment_per_100k"]
+    .mean()
+)
+tennessee_results
+fl_result = (
+    florida_prescriptions_result.groupby("Year")["shipment_per_100k"]
+    .mean()
+    .reset_index()
+)
+comp_result = (
+    other_states_result.groupby(["Year"])["shipment_per_100k"].mean().reset_index()
+)
+florida_summary = pd.DataFrame(fl_result.describe()["shipment_per_100k"]).rename(
+    columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Florida"}
+)
+comp_summary = pd.DataFrame(comp_result.describe()["shipment_per_100k"]).rename(
+    columns={"shipment_per_100k": "Opioid Shipment per 100k Residents - Control States"}
+)
+stats = pd.concat([florida_summary, comp_summary], axis=1)
+# create a scale for number of years before and after 2012 (target year)
+
+
+def scale_years(year):
+    if year == 2007:
+        return -3
+    if year == 2008:
+        return -2
+    if year == 2009:
+        return -1
+    if year == 2010:
+        return 0
+    if year == 2011:
+        return 1
+    if year == 2012:
+        return 2
+
+
+fl_result["year relative to policy"] = fl_result["Year"].apply(lambda x: scale_years(x))
+comp_result["year relative to policy"] = comp_result["Year"].apply(
+    lambda x: scale_years(x)
+)
+
+# double check no nulls in "year relative to policy"
+
+assert fl_result["year relative to policy"].isnull().sum() == 0
+assert comp_result["year relative to policy"].isnull().sum() == 0
```

Comment:
This code will need to be deleted, since we are not required to scale down the year values with respect to 2010 (for eg: 2007 - (-3), 2008 - (-2) and so on). We are supposed to show the years in the plots as they are - eg: 2007, 2008 and so on.

Reply:
Thanks for pointing out all the errors for me!